### PR TITLE
[web3]: remove `default` specification on types

### PR DESCRIPTION
**Pull request by dkent600**
This removes the `default` specification from all types that have it.  

Most or all of the major types in these web3 type definitions were defined as `default` types.  This is a problem:

1. `default` types cannot be augmented.  Augmentation is desireable when working around bugs in the typings and for any application-idiomatic need to extend the web3 API.
2. making some types default and other types not default make inconsistent the syntax for importing the two.

As far as I can tell, there is no need for these types to be defined as `default`.

I don't see that this breaks anything, and it enables the conventional TypeScript import syntax to work:

```
import { Web3 } from "web3";
```

The following is no longer the only option, but still works:

```
const Web3 = require('web3');
```

### CODE REVIEW

File: scripts/fix-tslint.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    return out;
+}
+function fixRules(rules) {
+    const out = {};
+    for (const key in rules) {
+        out[key] = rules[key];
+    }
+    return out;
+}
```
Comment (Levino) [inline scripts/fix-tslint.js]: What is this? Why do we need it?

File: scripts/generate-tsconfigs.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return out;
+}
+function fixCompilerOptions(config) {
+    const out = {};
+    for (const key in config) {
+        out[key] = config[key];
+        // Do something interesting here
+    }
+    return out;
+}
```
Comment (Levino) [inline scripts/generate-tsconfigs.js]: Same as above. Why is this necessary?

File: types/web3/index.d.ts [hunk 3]
```diff
@@ -50,5 +51,3 @@ declare class Web3 {
     setProvider(provider: Provider): void;
     utils: Utils;
 }
-
-export = Web3;
```
Comment (Levino) [inline types/web3/index.d.ts]: I fear this is important. Web3 uses `module.exports` https://github.com/ethereum/web3.js/blob/326e42fe3687d3e34840dbe4797408dfe6364917/packages/web3/src/index.js#L78

File: types/web3/web3-tests.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import Web3 = require("web3");
+import { Web3 } from "web3";
```
Comment (Levino) [inline types/web3/web3-tests.ts]: This might work for the type definitions but not for the library which uses `module.exports`. Maybe this works when using synthetic default exports in the tsconfig.json but even if, then I would assume the correct syntax is actually `import Web3 from 'web3'`. Did you verify that this actually works when using `web3` in a project?

**Review by levino — CHANGES_REQUESTED** [review]
Thank you for the work you put into this PR. I fear that we should have discussed this before you actually started on the work. I am not sure that it is always better to have named exports instead of default exports. This is something we should come to a conclusion first, before we move any further with this PR. Please create an issue like "Only use named exports in web3" and  elaborate why you want to remove the default exports. Also please address again in there or here why we can remove the toplevel `export = Web3`. I think that is not possible.

**Comment by dkent600** [comment]
@Levino Indeed I'm finding issues that I need to investigate further, but won't have time for another several weeks, so I will close this PR for the time-being.  However, I really hope we can find a solution that works for all the needed use cases while not requiring that we define `default` types.
